temporal-sdk/src/main/java/io/temporal/common/interceptors/ActivityInboundCallsInterceptor.java

```diff
@@ -22,7 +22,25 @@
 import io.temporal.activity.ActivityExecutionContext;
 
 public interface ActivityInboundCallsInterceptor {
+  final class ActivityInput {
+    private final Header header;
+    private final Object[] arguments;
+
+    public ActivityInput(Header header, Object[] arguments) {
+      this.header = header;
+      this.arguments = arguments;
+    }
+
+    public Header getHeader() {
+      return header;
+    }
+
+    public Object[] getArguments() {
+      return arguments;
+    }
+  }
+
   void init(ActivityExecutionContext context);
 
-  Object execute(Object[] arguments);
+  Object execute(ActivityInput input);
```

Comment:
This would introduce backwards incompatible change to the public interface and may break anyone who is already using this interceptor.
I think we should think about alternative options.

Reply:
I think it is OK to break this. I doubt it is widely used and the change is localized. I would be against a breaking change that affects the actual workflow code.

Reply:
If we are going to break this I would add `ActivityOutput` struct as well.

Reply:
addressed